docs/documentation-styleguide.md

```diff
@@ -0,0 +1,185 @@
+# Carbon Documentation Style Guide
+## Follow these guidelines when contributing documentation to the Carbon repository
+
+### Contents
+
+[Introduction](#introduction)
+
+[What We Document](#what-we-document)
+* [User Facing Documentation](#user-facing-documentation)
+* [Release Notes and Changelog](#release-notes-and-changelog)
+* [Request For Change (RFC)](#request-for-change)
```

Comment:
Could you double check that this RFC skip-link works?

Reply:
I missed off "-rfc" 😞 I added that afterwards and didn't update the link. Corrected.